transcom/mymove
# Int b 19814 add origin/dest sit auth end date to db

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19814)

## Summary

This Pull request adds the columns origin_sit_auth_end_date and dest_sit_auth_end_date to the mto_shipments table. These dates will be calculated as the day the shipment enters origin/dest SIT + the SIT day allowance (DOFSIT/DDFIST + sit_days_allowance). Storing these values as columns in the database will reduce the need of computing these values on the frontend/backend and instead allow them to be statically held in the db and allow Prime to request an endpoint for the values when needed. 

## Verification Steps for the Author

These are to be checked by the author.

- [v] Tested in the local environment
- [v] Have the Agility acceptance criteria been met for this change?
- [v] Columns are added to the mto_shipments table in the dev_db

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. run git fetch --all in terminal of mymove repo to fetch all updated branches.
2. git checkout integrationTesting
3. git pull
4. git checkout INT-B-19814-add-origin/dest-sit-auth-end-date-to-db to checkout the code from this PR 
5. make db_dev_run
6. make db_dev_fresh db_dev_migrate to run the newly added migration script
7. Once the scripts have ran open up dbeaver and connect to the database and navigate to the mto_shipments table and verify origin_sit_auth_end_date and dest_sit_auth_end_date have been added as columns to the table as a date datatype. 

### Database

#### Any new migrations/schema changes:


## Screenshots
<img width="1728" alt="Screenshot 2024-05-13 at 11 33 22 AM" src="https://github.com/transcom/mymove/assets/167351500/44bbe450-8f34-4447-9705-7ad49b23d549">
